WcaOnRails/app/mailers/competitions_mailer.rb

```diff
@@ -122,14 +122,21 @@ def results_submitted(competition, results_submission, submitter_user)
     @competition = competition
     @results_submission = results_submission
     @submitter_user = submitter_user
-    file_name = "Results_#{competition.id}_#{Time.now.utc.iso8601}.json"
-    attachments[file_name] = results_submission.results_json_str
+    last_uploaded_json = @competition.uploaded_jsons.last
+    if last_uploaded_json
+      attachments["Results for #{@competition.id}.json"] = {
+        mime_type: "application/json",
+        content: last_uploaded_json.json_str,
+      }
+    end
     mail(
       to: "results@worldcubeassociation.org",
       cc: competition.delegates.pluck(:email),
       reply_to: competition.delegates.pluck(:email),
       subject: "Results for #{competition.name}",
     )
+    # Cleanup the uploaded jsons now that we attached the relevant one when mailing the WRT.
+    @competition.uploaded_jsons.delete_all
```

Comment:
Have you confirmed that sending emails works? I had this vague concern that these mail methods expect the call to `mail` to be the very last thing in each method, but I could be crazy.

Reply:
I've tried it multiple times with mailcatcher to be able to send screenshots of the emails to the WRT.
Did you have something else in mind?